src/parser/xasl_generation.c

```diff
@@ -5733,6 +5733,8 @@ pt_make_regu_hostvar (PARSER_CONTEXT * parser, const PT_NODE * node)
 		{
 		  regu->domain->codeset = db_get_string_codeset (val);
 		  regu->domain->collation_id = db_get_string_collation (val);
+		  regu->domain->precision = db_value_precision (val);
```

Comment:
Do we need to have this kind of details (precision, scale) on the domain ?
If a second execution with a db value having precision > precision of previous execution value, the new value will be truncated.

Also, this leads to increase of number of cached domains which will slow down the search (this is not major issue, since this is client code).

Reply:
Yes, wrong precision was the bug. Should not be an issue, since the value was previously converted to expected domain, which does not change.